runtime-decompiler/src/main/java/org/jrd/frontend/frame/settings/SettingsView.java

```diff
@@ -55,43 +69,150 @@
 
             chooser = new JFileChooser();
             File dir;
-            if (isPortable()) {
-                dir = new File(getJrdLocation() + File.separator + "libs");
+            if (Directories.isPortable()) {
+                dir = new File(Directories.getJrdLocation() + File.separator + "libs");
             } else {
-                dir = new File(getJrdLocation() + File.separator + "decompiler_agent" + File.separator + "target");
+                dir = new File(Directories.getJrdLocation() + File.separator +
+                        "decompiler_agent" + File.separator + "target");
             }
-            chooser.setCurrentDirectory(fallback(dir));
+            chooser.setCurrentDirectory(FileSelectorArrayRow.fallback(dir));
 
             this.setLayout(new GridBagLayout());
             GridBagConstraints gbc = new GridBagConstraints();
             gbc.anchor = GridBagConstraints.WEST;
             gbc.fill = GridBagConstraints.BOTH;
 
-            gbc.insets = new Insets(0,20,0,0);
+            addAgent(gbc);
+
+            // Nested Jars
+            nestedJars = new JLabel("Nested Jars Settings:");
+            newExtensionsTextField = new JTextField();
+            newExtensionsTextField.addKeyListener(new KeyListener() {
+                @Override
+                public void keyTyped(KeyEvent keyEvent) {
+
+                }
+
+                @Override
+                public void keyPressed(KeyEvent keyEvent) {
+                    if (keyEvent.getKeyCode() == KeyEvent.VK_ENTER) {
+                        confirmExtensions();
+                    }
+                }
+
+                @Override
+                public void keyReleased(KeyEvent keyEvent) {
+
+                }
+            });
```

Comment:
See 
```java
.addKeyListener(new KeyAdapter() {
            @Override
            public void keyReleased(KeyEvent e) {
```
 in BytecodeDecompilerView. Please apply that knowledge (your own changeset from March btw lol) here.